lib/jormungandr/test/integration/Cardano/WalletSpec.hs

```diff
@@ -0,0 +1,154 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.WalletSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Launcher
+    ( Command (..), StdStream (..), launch )
+import Cardano.Wallet
+    ( BlockchainParameters (..)
+    , WalletLayer (..)
+    , newWalletLayer
+    , waitForWalletSync
+    )
+import Cardano.Wallet.Jormungandr.Compatibility
+    ( BaseUrl (..), Jormungandr, Scheme (..) )
+import Cardano.Wallet.Jormungandr.Environment
+    ( Network (..) )
+import Cardano.Wallet.Jormungandr.Network
+    ( JormungandrLayer (..)
+    , defaultManagerSettings
+    , getInitialBlockchainParameters
+    , newManager
+    )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Passphrase (..), digest, generateKeyFromSeed, publicKey )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( mkSeqState )
+import Cardano.Wallet.Primitive.Mnemonic
+    ( EntropySize, entropyToBytes, genEntropy )
+import Cardano.Wallet.Primitive.Model
+    ( currentTip )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..)
+    , Hash (..)
+    , SlotId (..)
+    , SlotLength (..)
+    , WalletId (..)
+    , WalletName (..)
+    )
+import Cardano.Wallet.Unsafe
+    ( unsafeRunExceptT )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( Async, async, cancel )
+import Control.Monad
+    ( unless, void )
+import Data.Coerce
+    ( coerce )
+import Data.Time.Clock
+    ( addUTCTime, getCurrentTime )
+import System.Directory
+    ( removePathForcibly )
+import Test.Hspec
+    ( Spec, after, before, expectationFailure, it )
+
+import qualified Cardano.Wallet.DB.MVar as MVar
+import qualified Cardano.Wallet.Jormungandr.Network as Jormungandr
+import qualified Data.ByteString as BS
+
+spec :: Spec
+spec = do
+    before startNode $ after killNode $ do
+        it "A newly created wallet can sync with the chain" $ \(_, wallet) -> do
+            bytes <- entropyToBytes <$> genEntropy @(EntropySize 15)
+            let xprv = generateKeyFromSeed (Passphrase bytes, mempty) mempty
+            wid <- unsafeRunExceptT $ createWallet wallet
+                (WalletId $ digest $ publicKey xprv)
+                (WalletName "My Wallet")
+                (mkSeqState @(Jormungandr 'Testnet) (xprv, mempty) minBound)
+            unsafeRunExceptT $ restoreWallet wallet wid
+            threadDelay (8 * second)
+            tip <- slotId . currentTip . fst <$>
+                unsafeRunExceptT (readWallet wallet wid)
+            unless (tip > (SlotId 0 0)) $
+                expectationFailure ("The wallet tip is still " ++ show tip)
+
+        -- This tests both that the wallet syncs correctly, and that
+        -- 'slotIdTime' computes the time of the wallet tip correctly.
```

Comment:
So this test is testing two things… We could just get the tip from the `NetworkLayer` and call `slotIdTime` on it. We could get by without any `waitForWalletSync`. On the other hand I thought it was kind of interesting to use the `currentTip`.